packages/emails/lib/generateIcsString.ts

```diff
@@ -0,0 +1,142 @@
+import type { DateArray, ParticipationStatus, ParticipationRole, EventStatus } from "ics";
+import { createEvent } from "ics";
+import type { TFunction } from "next-i18next";
+import { RRule } from "rrule";
+
+import dayjs from "@calcom/dayjs";
+import { getRichDescription } from "@calcom/lib/CalEventParser";
+import { getWhen } from "@calcom/lib/CalEventParser";
+import type { CalendarEvent, Person } from "@calcom/types/Calendar";
+
+export enum BookingAction {
+  Create = "create",
+  Cancel = "cancel",
+  Reschedule = "reschedule",
+  RequestReschedule = "request_reschedule",
+  LocationChange = "location_change",
+}
+
+const generateIcsString = ({
+  event,
+  t,
+  role,
+  bookingAction,
+}: {
+  event: CalendarEvent;
+  t: TFunction;
+  role: "attendee" | "organizer";
+  bookingAction: BookingAction;
+}) => {
+  let title = "",
+    subtitle = "",
+    status: EventStatus = "TENTATIVE";
+  // Taking care of recurrence rule
+  let recurrenceRule: string | undefined = undefined;
+  const partstat: ParticipationStatus = "ACCEPTED";
+  const icsRole: ParticipationRole = "REQ-PARTICIPANT";
+  if (event.recurringEvent?.count) {
+    // ics appends "RRULE:" already, so removing it from RRule generated string
+    recurrenceRule = new RRule(event.recurringEvent).toString().replace("RRULE:", "");
+  }
+
+  switch (bookingAction) {
+    case BookingAction.Create:
+      if (role === "organizer") {
+        title = event.recurringEvent?.count ? "new_event_scheduled_recurring" : "new_event_scheduled";
```

Comment:
Currently bookingScenario tests don't check the title in ics file but they do test it in Email content, we can quickly add the support to verify this in ics file. We would just need to modify `toHaveEmail` function as it already has all the appropriate data.

Reply:
We could do it as a followup as well if it takes more time just in case as there are unit tests for this function which adds some confidence already.

Reply:
For long term I think no, because with which params `generateIcsString` is called and whether it's called aren't tested. That can only be tested by integration tests or e2e tests.

But for now it's fine, we can work on #12214 later.